### PR TITLE
[wasm64] Cast BigInt argument to i64 Number

### PR DESCRIPTION
**Pull request by tiran**
Fix argument type and struct access of various functions on WASM64:

- system
- gethostaddr
- gethostbyname
- gethostbyname_r
- getaddrinfo
- getnameinfo
- connect
- bind
- sendto
- doStat (internal helper for stat calls)
- pwrite
- pread
- read return value

Closes: #17595
Closes: #17620

### CODE REVIEW

**Comment by tiran** [comment]
@kripken Another WSAM64 issue. The issue has a simple reproducer for the problem.

File: src/library.js [hunk 1]
```diff
@@ -1733,6 +1733,7 @@ mergeInto(LibraryManager.library, {
   $readSockaddr__deps: ['$Sockets', '$inetNtop4', '$inetNtop6', 'ntohs'],
   $readSockaddr: function (sa, salen) {
     // family / port offsets are common to both sockaddr_in and sockaddr_in6
+    {{{ from64('sa') }}}
```
Comment (sbc100) [inline src/library.js:1736]: The preferred way to do these conversions is by specifying signatures for the JS functions that take addresses.  In this case I think it means that easy of the library entry points that calls `readSockAddr` can/should specify a `__sig` attribute with `p` in the positions that correspond to pointer-width/size_t-width things.

File: src/library_wasi.js [hunk 2]
```diff
@@ -211,6 +211,7 @@ var WasiLibrary = {
   },
   $doWritev__docs: '/** @param {number=} offset */',
   $doWritev: function(stream, iov, iovcnt, offset) {
+    {{{ from64('iov') }}}
```
Comment (sbc100) [inline src/library_wasi.js:214]: Ditto.. by the time `doWritev` is called we expect iov to already have been converted by the `__sig` of the system calls.

**Comment by sbc100** [comment]
I would hope that along with this change we can enable more `wasm64l` tests..  normally with each such fix we increase the number of tests that can pass.

File: src/library.js (hunk 1 above)
Comment (tiran) [reply src/library.js]: Ah, I was not aware that `__sig` handles conversion, too. Thanks for pointing it out.

**Review by sbc100 — APPROVED** [review]

**Pull request merged**